blog/content/post/2021-10-10-optiplex-dasharo.md

```diff
@@ -0,0 +1,173 @@
+you may read about in [other blog posts](https://blog.3mdeb.com/2020/2020-06-24-dell-optiplex-port/)
+by the way) but it was definitely worth it. It opened a path for developing a
+firmware that would squeeze out the full potential of the platform. You may be
+wondering what else a platform may do that OEM didn't implement. But I will
+tell you this, with open source firmware almost anything is possible and with
+[Dasharo firmware](https://dasharo.com/) we make it a reality.
+
+## ME neutralization
+
+Intel Management Engine is a very controversial component of modern Intel
+chipsets and SoCs due to its capabilities and features. One of the most fearful
+features is the remote management and connectivity. Intel ME is able to utilize
+network via dedicated network controller on vPro machines without any awareness
+of the operating system software. It is used for software like Intel Active
+Management Technology ([AMT](https://www.intel.co.uk/content/www/uk/en/architecture-and-technology/intel-active-management-technology.html))
+or [Computrace](https://i.dell.com/sites/content/business/solutions/brochures/en/Documents/absolute-overview.pdf).
+Such features and capabilities often raises doubt and concerns about privacy.
+Moreover it is a few megabytes of proprietary closed source firmware of unknown
+quality with uncertain other impacts on the system. Many people often want to
+disable ME or even neutralize it to reduce the attack surface. This is what we
+offer with Dasharo firmware on Dell OptiPlex 7010 / 9010, [me_cleaner](https://github.com/corna/me_cleaner)
+is being applied on the firmware image which can be flashed on the board. From
+a 6MB of unknown firmware only dozens of kilobytes are left.
+
+## RAM support
+
+As engineers we often encounter situation that the same board comes in a few
+SKUs (stock keeping units) which may be distinguished with by the used
+processor for example. Such a procedure is often utilized to reduce the cost of
+the product or rather offer a product less or more equipped to make it more
+affordable on the market. Also as engineers doing open source firmware we want
+to know how many differences they are, whether we may unify support of the
+variants or even use the same binary to support multiple boards. But in the
+cases where the mainboard is essentially identical and the OEM offers it with
+different hardware capabilities, one question comes to the mind immediately:
+what is going on? It happens for the 7010 and 9010 variants in the [specsheet](https://www.dell.com/support/manuals/en-us/optiplex-7010/opti7010_usff/specifications?guid=guid-157e8495-34d3-4efa-ab61-1d9efba4c90e).
+
+```
+Maximum memory:
+  OptiPlex 7010	16 GB
+  OptiPlex 9010	32 GB
+```
+
+Why would maximum RAM be limited by the variant? The number of memory slots and
+the used processors is the same so there should not be any differences. It may
+be that the firmware limits the RAM capacity on the 7010 machine (very ugly)
```

Comment:
@miczyg1 did we proved it?

Reply:
Hard to prove, but easy to check... However, we didn't do it yet.

Reply:
Let's do it, maybe TAT can help.

Reply:
Still not checked, I may quickly find it out in the office tomorrow.

Reply:
Done the checks. 32GB RAM works on 7010 as well, no 16GB limitations or whatsoever.